HACKING.md

```diff
@@ -79,6 +79,79 @@ source .tox/unit/bin/activate
 pytest -v test/test_real_pebble.py
 ```
 
+## Using an `ops` branch in a charm
+
+When making changes to `ops`, you'll commonly want to try those changes out in
+a charm.
+
+### From a Git branch
+
+If your changes are in a Git branch, you can simply replace your `ops` version
+in `requirements.txt` (or `pyproject.toml`) with a reference to the branch, like:
+
+```
+#ops ~= 2.9
+git+https://github.com/{your-username}}/operator@{your-branch-name}}
```

Comment:
There are mismatched braces here.

Also, do we need the [`#egg=ops` thing](https://stackoverflow.com/questions/11835396/why-egg-foo-when-pip-installing-from-git-repo) at the end of the line here? I have always used it, but maybe it's not needed?

Reply:
> There are mismatched braces here.

Thanks, fixed!

> Also, do we need the [`#egg=ops` thing](https://stackoverflow.com/questions/11835396/why-egg-foo-when-pip-installing-from-git-repo) at the end of the line here? I have always used it, but maybe it's not needed?

Hmm, the plain version definitely works for me. But good call, I checked the [docs](https://pip.pypa.io/en/stable/topics/vcs-support/):

> pip also looks at the egg fragment specifying the “project name”. In practice the egg fragment is only required to help pip determine the VCS clone location in editable mode. In all other circumstances, the egg fragment is not necessary and its use is discouraged.

So since editable mode isn't relevant here, I believe the `#egg=` should not be included.

Reply:
Thanks, good to know.